kep/9-capacity-scheduling/README.md

```diff
@@ -0,0 +1,324 @@
+<= max <= hard` . 
+
+
+### User Stories (optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system.  The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+We assume two elastic quotas are defined: QuotaA (min:4, max:6) and QuotaB 
+(min:6, max:8), the quota unit is the number of GPUs. UserA in green color 
+consumes QuotaA and UserB in red color consumes QuotaB. The entire cluster 
+has 10 GPUs available hence the sum of quota min is equal to the cluster capacity.
+
+Initially, UserA consumes 2 GPUs and UserB consumes 3 GPUs. Both consumptions
+are within the reserved "min" quota range.
+
+![1](./1.png)
+
+
+Next, UserA consumes 4 GPUs reaching the "min" quota of QuotaA. UserB still 
+consumes 3 GPUs. The cluster still has sufficient GPU resources hence no 
+resource stealing happens.
+
+![2](./2.png)
+
+Assuming UserA attempts to use the rest of 3 unused GPUs in the cluster,
+only 2 more GPUs can be allocated to reach its quota "max", which is 6 GPUs.
+Note that in this case, 2 GPUs are "borrowed" from UserB's "Guaranteed" quota
+since UserB is not using them at that moment. 
+
+![3](./3.png)
+
+Later, if UserB demands 3 more GPUs compared to its current consumption, 
+its request cannot be immediately satisfied since only one unused GPU left in
+the cluster. Since UserB's "min" quota is 6 GPUs, the preemption algorithm 
+ensures that three GPUs will be returned back by preempting a pod which 
+requested 2 GPUs from UserA, i.e., from 6/3 to 4/6. Then if UserB requests 1 
+or 2 more GPUs, it won't succeed as GPUs in UserA by that moment are all 
+guaranteed and hence non-preemptable. 
+
+![4](./4.png)
+
+In the end, UserA consumes 4 GPUs and UserB consumes 6 GPUs, satisfying each 
+user's "min" quota.
+
+## Design Details
+
+### Extention point
+
+#### PreFilter
+We will check if the (Pod.request + Quota.Allocated) is less than Quota.Max. 
+If not, the scheduling cycle of PodA will fail. 
+
+#### PostFilter
+We will reimplement the method `selectVictimsOnNode` in `defaultPreempt`. The
+original `selectVictimsOnNode` method selects all the pods with the lower
+priority than the preemptor’s priority as potential victims in a node. 
+
+In this proposal, the potential victims selection logic needs to be changed 
+in the following cases.
+
+- if Preemptor.Request + Quota.allocated <= Quota.min: It means that its min 
+or guaranteed resource is used or `borrowed` by other Quota. Potential 
+victims in a node will be chosen from Quotas that allocates more resources 
+than its min, i.e., borrowing resources from other Quotas.
+- if Preemptor.Request + Quota.allocated > Quota.min: It means that its 
+guaranteed isn't borrowed by other quotas. So that we will select the pods 
+belongs to the same quota(namespace) with the lower priority than the 
```

Comment:
Today with regular quotas I don't think the scheduler necessarily restrict to the namespace when picking victims, should this rule also apply here?